### ISSUE TITLE
CPU usage on upload

### ISSUE THREAD

**Issue by nugaon** [ethersphere/bee#4282]
⚠️ Support requests in an issue-format will be closed immediately. For support, go to Swarm's [Discord](https://discord.gg/XGNBEKktkD).

### Context

Bee 1.17.3

### Summary

The CPU runs until its limits if one file was uploaded. The log is flooded various push/pull related problems:

```
"time"="2023-09-01 10:20:39.725002" "level"="debug" "logger"="node/pushsync" "msg"="no peers left" "chunk_address"="f8365bfeec632d642f3679c3bb418d5ea4c27b95a19d695588630bef34877f78" "error"="no peer found"
"time"="2023-09-01 10:20:39.742655" "level"="error" "logger"="node/pusher" "msg"="pusher: failed PushChunkToClosest" "error"="no peer found"
"time"="2023-09-01 10:20:39.742740" "level"="debug" "logger"="node/libp2p" "msg"="handle protocol failed" "protocol"="pullsync" "version"="1.3.0" "stream"="pullsync" "peer"="48013b5c5142e482759f08ef60057b4bbb7cb2e77790b44e3f06b235493347d4" "error"="make offer: context canceled"
"time"="2023-09-01 10:20:39.738617" "level"="debug" "logger"="node/puller" "msg"="syncWorker interval failed" "peer_address"="1ec1e220b1f6ab66005fb8308fb97f73174a1eaabe18b39158cea52a48143e08" "bin"=13 "cursor"=0 "start"=1 "topmost"=0 "err"="new stream: send headers: read message: context canceled"
```

### Expected behavior

Possibility to upload data with reasonable amount of log messages and CPU usage.

### Actual behavior

I had to stop the my small cluster after couple of minutes because of the CPU heat.

### Steps to reproduce

```
fdp-play start --bee-version 1.17.3
swarm-cli stamp buy --depth 27 --amount 10M
swarm-cli upload [filepath]
```

**Comment by tmm360** [comment]
I also have noticed an unjustified cpu usage after an upload, I'm working in mainnet. CPU doesn't seam to be used well (it looks like a single thread process), and it has been blocked for more than 12h now with a single pending upload, without client is able to send new bits.
Screenshot:
![image](https://github.com/ethersphere/bee/assets/7620704/1da82ca0-c028-4d77-8fd6-2a7e4d1ad7ce)

**Comment by tmm360** [comment]
@nugaon maybe this issue should be addressed to a better cpu usage? I don't know... I've tried now to stop the pending upload, trying to see if something will resolve.

**Comment by tmm360** [comment]
37h now, still busy
![image](https://github.com/ethersphere/bee/assets/7620704/4739c59e-58c0-419b-9441-6f298d837d39)

**Comment by istae** [comment]
@nugaon looks like your node has a connection problem and is unable to push chunks out to peers. having no peers means the pusher fails very quickly and retries immediately again, causing high load on the CPU.

**Comment by tmm360** [comment]
@istae mine 2 nodes has poor connectivity now, but this wasn't the case when issue started.

This is last week of connectivity depth (node 0 and 1):
![image](https://github.com/ethersphere/bee/assets/7620704/1df07d99-1dd1-4061-a5fb-57c08baccfb2)

Last week of CPU on node 0 (no problems):
![image](https://github.com/ethersphere/bee/assets/7620704/32bb5361-22af-4fda-8c30-9ea8750573c4)

Last week of CPU on node 1 (with problem):
![image](https://github.com/ethersphere/bee/assets/7620704/43b9137d-dc57-4693-a229-355432d79d9a)

**Comment by tmm360** [comment]
With a node restart depth has been recovered, now it's back to 9, but CPU is still high as if nothing changed.

**Comment by tmm360** [comment]
4 days now, connection depth is 9, CPU is still high loaded, and node is not able to receive any new upload from bzz (didn't tried other methods). It looks like input stream is totally closed, because client fails with error "Error while copying content to a stream.", and this suggest me that it's failing still opening the stream itself.

**Comment by nugaon** [comment]
> @nugaon looks like your node has a connection problem and is unable to push chunks out to peers. having no peers means the pusher fails very quickly and retries immediately again, causing high load on the CPU.

I used this very same setup for my whole JS development on Bee, all clients run on my laptop with the same connection.

**Comment by asabya** [comment]
Using this same setup I'm getting this
```
2023-09-07 08:08:07 "time"="2023-09-07 02:38:07.848238" "level"="debug" "logger"="node/libp2p" "msg"="stream handler: successfully connected to peer (inbound)" "address"="676790fcae312292ffc17b7f7c84d5b9acc51ef0ea3f27d0ff3bada3362abc5d" "light"="" "user_agent"=" bee/1.17.3-44846bba go1.20.7 linux/amd64"
2023-09-07 08:08:07 "time"="2023-09-07 02:38:07.891358" "level"="debug" "logger"="node/kademlia" "msg"="reachability of peer updated" "peer_address"="676790fcae312292ffc17b7f7c84d5b9acc51ef0ea3f27d0ff3bada3362abc5d" "reachability"="Public"
2023-09-07 08:08:26 "time"="2023-09-07 02:38:26.292766" "level"="info" "logger"="node/storageincentives" "msg"="entered new phase" "phase"="claim" "round"=7 "block"=1147
2023-09-07 08:08:26 "time"="2023-09-07 02:38:26.387450" "level"="error" "logger"="node/storageincentives" "msg"="phase failed" "error"="IsPlaying: overlay 0x0292b7ca8fe7d745339cbd05c19da0e2ba0fab18b63594b0029738a01af5d114 depth 0: VM Exception while processing transaction: revert stake amount does not meet minimum" "phase"="sample" "round"=7
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.470033" "level"="debug" "logger"="node/pushsync" "msg"="could not push to peer" "chunk_address"="6474fdaf8bf2bc0af33b9154a444ca712332d9b62b36bc76afe6635b09bbc9d0" "peer_address"="676790fcae312292ffc17b7f7c84d5b9acc51ef0ea3f27d0ff3bada3362abc5d" "error"="stream reset"
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.470275" "level"="debug" "logger"="node/pushsync" "msg"="no peers left" "chunk_address"="6474fdaf8bf2bc0af33b9154a444ca712332d9b62b36bc76afe6635b09bbc9d0" "error"="no peer found"
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.470341" "level"="error" "logger"="node/pusher" "msg"="pusher: failed PushChunkToClosest" "error"="no peer found"
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.470454" "level"="debug" "logger"="node/api/post_soc" "msg"="done split failed" "error"="no peer found"
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.470691" "level"="error" "logger"="node/api/post_soc" "msg"="done split failed"
2023-09-07 08:08:33 "time"="2023-09-07 02:38:33.471803" "level"="info" "logger"="node/api" "msg"="api access" "ip"="172.18.0.1" "method"="POST" "host"="localhost:1633" "uri"="/soc/86900e89e52798d410eba4c328412ad470905498/a715e336b403718e3a5f6864009e946c6647ffe8a3ff5855fc84b5822003a3b7?sig=50d5a8ff1cb76a2625bfa54bec4bd596edf5da2fbba856a6b4c4cb0ef2e9357622d26ad93ccf0ae006143c40c4cb635a49d39e592f05991db38f93e97e6ce2951c" "proto"="HTTP/1.1" "status"=500 "size"=68 "duration"="28.273791ms" "user-agent"="Go-http-client/1.1"
```

Response from the server `done split failed` with `500`

**Comment by tmm360** [comment]
After 12 days condition didn't change. CPU is still high as before. Connection depth is 8.

**Comment by tmm360** [comment]
Upgrade to 1.17.4 finally solved the cpu usage problem. Probably localstorage migration did the magic.

**Comment by tmm360** [comment]
Hi, unfortunately the last release 1.17.5 didn't solved for me. Both nodes has still high cpu load even after 1h from the update. @istae.
How can I provide you advanced logs?

**Comment by ldeffenb** [comment]
>How can I provide you advanced logs?

--verbosity 5

**Comment by tmm360** [comment]
Logs of ~15m.
[logs.zip](https://github.com/ethersphere/bee/files/12921627/logs.zip)

**Comment by istae** [comment]
i don't see ANY upload activity from your node, just network pushsync traffic.

**Comment by tmm360** [comment]
My fault. I didn't express myself very well, and maybe I would have to open another issue.
My problem STARTED the first time with an upload. After it, it persists until now, even after reboot.

![image](https://github.com/ethersphere/bee/assets/7620704/c37dd423-6a9c-47b8-ac45-31200251e74d)
Last 2 weeks of node activity, without any upload. I've disabled new uploads reception because it is very more slow and prone to fail uploads.

Problem is that more recently also my second node has started to do it. I can't say if related to an upload or not.
![image](https://github.com/ethersphere/bee/assets/7620704/120cbf6a-c007-4da2-b109-edcf4c53ff17)

**Comment by istae** [comment]
@tmm360 are you doing parallel uploads?
can you post logs from the `pusher` pkg?

**Comment by istae** [comment]
you can grep for pusher but you can also simply get the logs of a node thats busy with uploading for a 10-15 min

**Comment by tmm360** [comment]
I've tried to perform an upload, but each time it fails after some seconds, and without print bzz POST logs.
With this logs I've purchased successfully a postage batch, and I've tried to upload 10 times with it a file, failing each time.

Anyway, also the bug of high CPU usage was present for all the time. As I said, it happen now also in idle normally.

[logs.zip](https://github.com/ethersphere/bee/files/13209780/logs.zip)
